symphony/lib/toolkit/class.entrymanager.php

```diff
@@ -174,39 +174,48 @@ public static function edit(Entry $entry)
                 continue;
             }
 
+            $did_lock = false;
             try {
+                $did_lock = Symphony::Database()->query('LOCK TABLES tbl_entries_data_' . $field_id . ' WRITE');
                 Symphony::Database()->delete('tbl_entries_data_' . $field_id, sprintf("
                     `entry_id` = %d", $entry->get('id')
                 ));
-            } catch (Exception $e) {
-                // Discard?
-            }
 
-            if (!is_array($field) || empty($field)) {
-                continue;
-            }
+                if (!is_array($field) || empty($field)) {
+                    if ($did_lock) {
+                        Symphony::Database()->query('UNLOCK TABLES');
+                    }
+                    continue;
+                }
 
-            $data = array(
-                'entry_id' => $entry->get('id')
-            );
+                $data = array(
+                    'entry_id' => $entry->get('id')
+                );
 
-            $fields = array();
+                $fields = array();
 
-            foreach ($field as $key => $value) {
-                if (is_array($value)) {
-                    foreach ($value as $ii => $v) {
-                        $fields[$ii][$key] = $v;
+                foreach ($field as $key => $value) {
+                    if (is_array($value)) {
+                        foreach ($value as $ii => $v) {
+                            $fields[$ii][$key] = $v;
+                        }
+                    } else {
+                        $fields[max(0, count($fields) - 1)][$key] = $value;
                     }
-                } else {
-                    $fields[max(0, count($fields) - 1)][$key] = $value;
                 }
-            }
 
-            foreach ($fields as $index => $field_data) {
-                $fields[$index] = array_merge($data, $field_data);
+                foreach ($fields as $index => $field_data) {
+                    $fields[$index] = array_merge($data, $field_data);
+                }
+
+                Symphony::Database()->insert($fields, 'tbl_entries_data_' . $field_id);
+            } catch (Exception $e) {
+                Symphony::Log()->pushExceptionToLog($e, true);
             }
 
-            Symphony::Database()->insert($fields, 'tbl_entries_data_' . $field_id);
+            if ($did_lock) {
```

Comment:
This block *should* be in a finally statement to be 100% kasher (if I am not wrong?)